diff --git a/gdk/win32/gdksurface-win32.c b/gdk/win32/gdksurface-win32.c
index e9b2ce9552..11e5f0a4d5 100644
--- a/gdk/win32/gdksurface-win32.c
+++ b/gdk/win32/gdksurface-win32.c
@@ -4497,8 +4497,8 @@ _gdk_win32_surface_compute_size (GdkSurface *surface)
         }
       else
         {
-          size_changed = width != impl->next_layout.configured_width ||
-                         height != impl->next_layout.configured_height;
+          size_changed = surface->width != impl->next_layout.configured_width ||
+                         surface->height != impl->next_layout.configured_height;
 
           surface->width = impl->next_layout.configured_width;
           surface->height = impl->next_layout.configured_height;
